config/reader.go

```diff
@@ -150,15 +150,6 @@ func readCertificateDataFromCloudGRPC(ctx context.Context,
 		return tlsConfig{}, err
 	}
 
-	if !signalingInsecure {
-		if res.TlsCertificate == "" {
-			return tlsConfig{}, errors.New("no TLS certificate yet from cloud; try again later")
-		}
-		if res.TlsPrivateKey == "" {
-			return tlsConfig{}, errors.New("no TLS private key yet from cloud; try again later")
-		}
-	}
-
```

Comment:
nit: I think we may as well leave this block in here without the check for `!signalingInsecure`, I think it's logically equivalent and ends up being more compartmentalized

Reply:
Good idea! 9c7d9337d